feat: add Edly client branding settings to SiteConfiguration

Waleed-Mujahid opened the pull request:
- Introduced a new JSONField for 'edly_client_branding_and_django_settings' in the SiteConfiguration model.
- Updated the SiteConfigurationAdmin to include the new field in the admin interface.
- Added a method to retrieve configuration values from the new field with error logging.

This enhancement allows for better customization of the Edly client theme settings.

Waleed-Mujahid committed:
feat: add Edly client branding settings to SiteConfiguration

- Introduced a new JSONField for 'edly_client_branding_and_django_settings' in the SiteConfiguration model.
- Updated the SiteConfigurationAdmin to include the new field in the admin interface.
- Added a method to retrieve configuration values from the new field with error logging.

This enhancement allows for better customization of the Edly client theme settings.

Waleed-Mujahid committed:
feat: add edx_org_short_name field to SiteConfiguration model

- Introduced a new CharField 'edx_org_short_name' to the SiteConfiguration model with unique constraints and help text for better organization identification.
- Updated SiteConfigurationAdmin to include the new field in the admin interface.
- Added a migration to create the new field in the database.

This enhancement improves the ability to uniquely identify organizations within the system.

Waleed-Mujahid committed:
fix: correct filtering logic in filter_visible function

- Updated the filter_visible function to apply the _filter_visible_course_certificates directly to the queryset instead of filtering for non-null course credentials first.
- we do not generate course cert against a program on credentials

Waleed-Mujahid committed:
feat: fix program certificate visibilty issue

credentials check if the certifciates are earned for course runs for a program however we do not want this. In our case program cert is only generated when user completes all courses on lms

muhammadali286 commented on credentials/apps/core/models.py:
```diff
@@ -124,6 +138,13 @@ class SiteConfiguration(models.Model):
+        help_text=_('JSON string containing edly client theme branding & Django settings.'),
+        null=False,
+        blank=False,
+        default={}
```
Using a mutable default (default={}) is unsafe because the same dict instance can be shared across all model instances.
https://docs.djangoproject.com/en/5.2/ref/models/fields/#jsonfield

Waleed-Mujahid replied:
updated to `dict` as per documentation

muhammadali286 commented on credentials/apps/core/models.py:
```diff
@@ -124,6 +138,13 @@ class SiteConfiguration(models.Model):
+    edly_client_branding_and_django_settings = JSONField(
+        verbose_name=_('Edly client theme branding & Django settings'),
+        help_text=_('JSON string containing edly client theme branding & Django settings.'),
+        null=False,
+        blank=False,
+        default={}
```
Default value already guarantees a non-null default. So `null=False` is fine, but `blank=False` is redundant in this context.

Waleed-Mujahid replied:
removed

Waleed-Mujahid committed:
feat: update default value for edly_client_branding_and_django_settings field

- Changed the default value of the 'edly_client_branding_and_django_settings' JSONField in the SiteConfiguration model from an empty dictionary to the dict constructor for better clarity.
- Added a migration to reflect this change in the database schema.

This update enhances the model's configuration handling.
